sql/sql_table.cc

```diff
@@ -12007,57 +12061,13 @@ copy_data_between_tables(THD *thd, TABLE *from, TABLE *to, bool ignore,
     to->auto_increment_field_not_null= FALSE;
     if (unlikely(error))
     {
-      if (to->file->is_fatal_error(error, HA_CHECK_DUP))
-      {
-        /* Not a duplicate key error. */
-	to->file->print_error(error, MYF(0));
-        error= 1;
-	break;
-      }
+      if (copy_data_error_handler(error, ignore, to, thd, alter_ctx))
+        break;
       else
       {
```

Comment:
The `else` is redundant. We could just write
```c++
if (copy_data_error_handler(error, ignore, to, thd, alter_ctx)))
  break;
DBUG_ASSERT(ignore);
to->file->restore_auto_increment(prev_insert_id);
delete_count++;
```
Above, I would also replace the comment with an assertion.

Even better might be to rename the function and invert its return value to make it self-explanatory:
```c++
if (!copy_data_error_ignore(error, ignore, to, thd, alter_ctx)))
  break;
```